### PR TITLE
Make the docker cli overridable during build.

### PR DESCRIPTION
**Pull request by amwat**
This enables building with podman.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -33,7 +33,7 @@ COMMIT?=$(shell git rev-parse HEAD 2>/dev/null)
 LD_FLAGS:=-X sigs.k8s.io/kind/pkg/cmd/kind/version.GitCommit=$(COMMIT)
 # the output binary name, overridden when cross compiling
 KIND_BINARY_NAME?=kind
-
+DOCKER?=docker
```
Comment (BenTheElder) [inline Makefile]: We should comment here hinting that you might set this to podman for building

Comment (amwat) [reply Makefile]: done

**Comment by amwat** [comment]
xref: https://github.com/kubernetes-sigs/kind/issues/154

**Comment by BenTheElder** [comment]
/lgtm
/approve
Thanks!

**Comment by BenTheElder** [comment]
```
export DOCKER=podman
make build
```
whoop!

**Pull request merged**